Add MIX roster extension

lnjX opened the pull request.

badlop commented:
The xmpp library doesn't have tests, so I've run ejabberd's test suite using this PR, and it works correctly: https://github.com/badlop/ejabberd/actions/runs/1003732890

lnjX commented:
It's the roster part (`urn:xmpp:mix:roster:0`) of [MIX-PAM (XEP-0405)](https://xmpp.org/extensions/xep-0405.html) in the current version 0.5, but I think the roster part of MIX-PAM hasn't changed since XEP-0405 v0.1 anyway.

The new stanza namespaces are for MIX-CORE v0.14.2 or later. The stanzas haven't changed in the XEP, only the MIX submission ID has been dropped and archiving got optional.

---

I just rebased this and fixed one mistake: the submission ID doesn't have the new namespace now

Neustradamus commented:
To look easily:
- 0.1.0 to 0.1.1: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.1.0.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.1.1.html
- 0.1.1 to 0.2.0: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.1.1.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.2.0.html
- 0.2.0 to 0.2.1: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.2.0.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.2.1.html
- 0.2.1 to 0.3.0: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.2.1.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.3.0.html
- 0.3.0 to 0.3.1: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.3.0.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.3.1.html
- 0.3.1 to 0.4.0: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.3.1.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.4.0.html
- 0.4.0 to 0.4.1: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.4.0.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.4.1.html
- 0.4.1 to 0.5.0: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.4.1.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.5.0.html
- 0.5.0 to 0.5.1: http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0405-0.5.0.html&newfile=https://xmpp.org/extensions/attic/xep-0405-0.5.1.html
